src/lib/y2common_criteria/dialogs/installation.rb

```diff
@@ -0,0 +1,179 @@
+          "<p>Notice that you will have to enter the correct password each time you boot " \
+          "the system. So make sure to not lose it!</p>"
+        )
+      end
+
+      # Handler for the 'next' event (button)
+      def next_handler
+        return unless valid?
+
+        # put libgcrypt into FIPS mode. libstorage-ng calls the cryptsetup
+        # external command so this takes effect when formatting luks volumes.
+        # Needs to happen in instsys. The target system boots with fips=1 kernel
+        # command line so this setting is not needed.
+        Yast::WFM.Execute(Yast::Path.new(".local.mkdir"), "/etc/gcrypt")
+        Yast::WFM.Write(Yast::Path.new(".local.string"), "/etc/gcrypt/fips_enabled", "1")
+
+        write_passphrase
+
+        super
+      end
+
+      protected
+
+      # @return [Y2Storage::EncryptPasswordChecker]
+      attr_reader :passwd_checker
+
+      # Introductory text to explain how the role works
+      #
+      # @return [String]
+      def description_text
+        _(
+          "<p>Please refer to the deployment guide before proceeding with the " \
+          "installation of this Common Criteria evaluated configuration.</p>\n" \
+          "<p>Enter a passphrase below to be used by default when encrypting devices " \
+          "during system installation.</p>\n" \
+          "<p>Read Help for more details.</p>\n"
+        )
+      end
+
+      # @return [Yast::Term] ui content for dialog
+      def passwd_widget(id, label)
+        Password(Id(id), Opt(:hstretch), label, proposal_features["encryption_password"] || "")
+      end
+
+      # Whether the information entered in the form is acceptable
+      #
+      # @return [Boolean]
+      def valid?
+        valid_passphrase? && good_passphrase?
+      end
+
+      # @see #valid?
+      def valid_passphrase?
+        msg = passwd_checker.error_msg(widget_value(:passphrase), widget_value(:repeat_passphrase))
+        return true if msg.nil?
+
+        Yast::Report.Warning(msg)
+        false
+      end
+
+      # @see #valid?
+      #
+      # User has the last word to decide whether to use a weak passphrase.
+      def good_passphrase?
+        message = passwd_checker.warning_msg(widget_value(:passphrase))
+        return true if message.nil?
+
+        popup_text = message + "\n\n" + _("Really use this passphrase?")
+        Yast::Popup.AnyQuestion(
+          "",
+          popup_text,
+          Yast::Label.YesButton,
+          Yast::Label.NoButton,
+          :focus_yes
+        )
+      end
+
+      # Helper to get the value of the given widget
+      def widget_value(id)
+        Yast::UI.QueryWidget(Id(id), :Value)
+      end
+
+      # Writes the entered passphrase into the default storage proposal settings
+      def write_passphrase
+        proposal = proposal_features
+        proposal["encryption_password"] = widget_value(:passphrase)
+        log.info "Writing CC passphrase at default storage proposal settings"
```

Comment:
Question: are proposal settings logged somewhere? If so, maybe we should not use a string for the password.

Reply:
When the proposal settings are logged, this method is used[1] so we are safe on that front. But maybe the ProductFeatures are dumped at some later point. I would need to check.

[1] https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/proposal_settings.rb#L312

Reply:
Fixed by feb23e6808aa6aa3217acb7530110f243aeebc04